components/admin_console/admin_definition.jsx

```diff
@@ -0,0 +1,599 @@
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import {Constants} from 'utils/constants';
+import {invalidateAllCaches, reloadConfig} from 'actions/admin_actions';
+
+import {CONNECTION_SECURITY_HELP_TEXT_WEBSERVER, WEBSERVER_MODE_HELP_TEXT} from './admin_definition_constants';
+
+export default {
+    reporting: {
+    },
+    settings: {
+        general: {
+            configuration: {
+                schema: {
+                    id: 'ServiceSettings',
+                    name: 'admin.general.configuration',
+                    name_default: 'Configuration',
+                    settings: [
+                        {
+                            type: Constants.SettingsTypes.TYPE_BANNER,
+                            label: 'admin.rate.noteDescription',
+                            label_default: 'Changing properties other than Site URL in this section will require a server restart before taking effect.',
+                            banner_type: 'info',
+                        },
+                        {
+                            type: Constants.SettingsTypes.TYPE_TEXT,
+                            key: 'SiteURL',
+                            label: 'admin.service.siteURL',
+                            label_default: 'Site URL:',
+                            help_text: 'admin.service.siteURLDescription',
+                            help_text_default: 'The URL that users will use to access Mattermost. Standard ports, such as 80 and 443, can be omitted, but non-standard ports are required. For example: http://mattermost.example.com:8065. This setting is required.',
+                            placeholder: 'admin.service.siteURLExample',
+                            placeholder_default: 'Ex "https://mattermost.example.com:1234"',
+                        },
```

Comment:
Could we get a comment at the top of this file detailing the different types and then what all the possible types/fields are?

Reply:
Sure, actually I can extract from the tests where I have defined the schema for this :)